features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Subscriptions/Subscriptions.jsx

```diff
@@ -53,8 +57,40 @@ function Subscriptions(props) {
     const { api, updateAPI } = props;
     const restApi = new API();
     const [tenants, setTenants] = useState(null);
+    const [policies, setPolices] = useState({});
+    const [availability, setAvailability] = useState({});
+    const [tenantList, setTenantList] = useState([]);
     const [subscriptions, setSubscriptions] = useState(null);
+    const [updateInProgress, setUpdateInProgress] = useState(false);
+
+    /**
+     * Save subscription information (policies, subscriptionAvailability, subscriptionAvailableTenants)
+     */
+    function saveAPI() {
+        setUpdateInProgress(true);
+        const { subscriptionAvailability } = availability;
+        const newApi = {
+            policies,
+            subscriptionAvailability,
+            subscriptionAvailableTenants: tenantList,
+        };
+        updateAPI(newApi)
+            .then(() => {
+                Alert.info('Subscription configurations updated successfully');
+            })
+            .catch((error) => {
+                console.error(error);
+                Alert.error('Error occurred while updating subscription configurations');
+            }).finally(() => {
+                setUpdateInProgress(false);
+            });
+    }
+
     useEffect(() => {
+        if (api) {
+            setAvailability({ subscriptionAvailability: api.subscriptionAvailability });
```

Comment:
why we set this inside an effect??

Reply:
I want to set it once the component is loaded. can I set this when I am defining the state variable instead?